### PR TITLE
mesos: 1.1.1 -> 1.4.0

### PR DESCRIPTION
**Pull request by disassembler**
###### Motivation for this change
mesos failed to build because of mismatch of automake version. Added autoreconfHook and moved some dependencies to nativeBuildInputs.

Related to #28643

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by disassembler** [comment]
nox-review is running currently. I'll check output in the morning.

**Comment by disassembler** [comment]
after rebase against master, mesos doesn't build. Working on fixing it.

**Comment by Mic92** [comment]
Feel free to change the title back, when mesos builds again.

File: pkgs/applications/networking/cluster/mesos/default.nix [hunk 1]
```diff
@@ -41,8 +41,14 @@ in stdenv.mkDerivation rec {
     ./nixos.patch
   ];
 
+  nativeBuildInputs = [
+    autoreconfHook
+    autoconf
+    automake
+    libtool
```
Comment (globin) [inline pkgs/applications/networking/cluster/mesos/default.nix]: You don't need `autoconf`, `automake` and `libtool` when including `autoreconfHook`

**Review by globin — CHANGES_REQUESTED** [review]

**Comment by disassembler** [comment]
Successful Build! Pinging maintainers @cstrahan @kevincox @offlinehacker for testing and review.

**Comment by globin** [comment]
I've just tried running the NixOS test, which failed.. I fixed the python handling in there, which makes it run further but fails with an error we had earlier (https://github.com/NixOS/nixpkgs/pull/19064)

I have no other way to test this..

**Comment by kamilchm** [comment]
Python framework failures were casued by dynamically linked `protobuf`. How about restoring protobuf substitues?

**Comment by disassembler** [comment]
doesn't build when statically linked:

```
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::LogMessage::operator<<(char const*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::AssignDescriptors(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::internal::MigrationSchema const*, google::protobuf::Message const* const*, unsigned int const*, google::protobuf::MessageFactory*, google::protobuf::Metadata*, google::protobuf::EnumDescriptor const**, google::protobuf::ServiceDescriptor const**)'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::SerializeAsString[abi:cxx11]() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormat::SerializeUnknownFields(google::protobuf::UnknownFieldSet const&, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteInt64(int, long, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::SerializeToString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::UnknownFieldSet::AddVarint(int, unsigned long)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteBool(int, bool, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `vtable for google::protobuf::internal::FunctionClosure0'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::ReadVarint64Fallback()'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormat::SkipField(google::protobuf::io::CodedInputStream*, unsigned int, google::protobuf::UnknownFieldSet*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::VerifyUtf8String(char const*, int, google::protobuf::internal::WireFormatLite::Operation, char const*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::CheckTypeAndMergeFrom(google::protobuf::MessageLite const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::FieldDescriptor::kTypeToCppTypeMap'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::IncrementRecursionDepthAndPushLimit(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::Int32Size(google::protobuf::RepeatedField<int> const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormat::ReadPackedEnumPreserveUnknowns(google::protobuf::io::CodedInputStream*, unsigned int, bool (*)(int), google::protobuf::UnknownFieldSet*, google::protobuf::RepeatedField<int>*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::LogMessage::LogMessage(google::protobuf::LogLevel, char const*, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::ReadVarint32Fallback(unsigned int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteUInt64(int, unsigned long, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteStringMaybeAliased(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::RegisterAllTypes(google::protobuf::Metadata const*, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::InitProtobufDefaults()'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::GetTypeName[abi:cxx11]() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::DiscardUnknownFields()'
./.libs/libmesos.so: undefined reference to `google::protobuf::EnumDescriptor::FindValueByNumber(int) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::ParseFromZeroCopyStream(google::protobuf::io::ZeroCopyInputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::Arena::OnArenaAllocation(std::type_info const*, unsigned long) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::ParsePartialFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
./.libs/libmesos.so: undefined reference to `typeinfo for google::protobuf::Message'
./.libs/libmesos.so: undefined reference to `google::protobuf::UnknownFieldSet::ClearFallback()'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::LogMessage::~LogMessage()'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::InitializationErrorString[abi:cxx11]() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedOutputStream::WriteStringWithSizeToArray(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned char*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::ArenaImpl::AddCleanup(void*, void (*)(void*))'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::ReadVarintSizeAsIntFallback()'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteInt32(int, int, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormat::SerializeUnknownFieldsToArray(google::protobuf::UnknownFieldSet const&, unsigned char*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::SerializeWithCachedSizesToArray(unsigned char*) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::SerializePartialToString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::UnknownFieldSet::MergeFrom(google::protobuf::UnknownFieldSet const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::FieldDescriptor::enum_type() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::InternalExtend(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::Reserve(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::PushLimit(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteBytesMaybeAliased(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::VerifyVersion(int, int, char const*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::PrintDebugString() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteDouble(int, double, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::UInt32Size(google::protobuf::RepeatedField<unsigned int> const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::ArenaImpl::AllocateAlignedAndAddCleanup(unsigned long, void (*)(void*))'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::OnShutdownDestroyMessage(void const*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteUInt32(int, unsigned int, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::DecrementRecursionDepthAndPopLimit(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::ArrayInputStream::ArrayInputStream(void const*, int, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::FunctionClosure0::~FunctionClosure0()'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::DebugString[abi:cxx11]() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::EnumDescriptor::FindValueByName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::ReadLittleEndian64Fallback(unsigned long*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteMessageMaybeToArray(int, google::protobuf::MessageLite const&, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageFactory::InternalRegisterGeneratedFile(char const*, void (*)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&))'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::ParseNamedEnum(google::protobuf::EnumDescriptor const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::BytesUntilLimit() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::SerializeToFileDescriptor(int) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::Descriptor::FindFieldByName(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::RepeatedPtrFieldBase::CloseGap(int, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::ReflectionOps::Merge(google::protobuf::Message const&, google::protobuf::Message*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteEnum(int, int, google::protobuf::io::CodedOutputStream*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::OnShutdownDestroyString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::MessageLite::ParseFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::fixed_address_empty_string[abi:cxx11]'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::LogFinisher::operator=(google::protobuf::internal::LogMessage&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::ReadBytes(google::protobuf::io::CodedInputStream*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::Message::SpaceUsedLong() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::ArenaImpl::AllocateAligned(unsigned long)'
./.libs/libmesos.so: undefined reference to `google::protobuf::GoogleOnceInitImpl(long*, google::protobuf::Closure*)'
./.libs/libmesos.so: undefined reference to `google::protobuf::DescriptorPool::InternalAddGeneratedFile(void const*, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormat::ComputeUnknownFieldsSize(google::protobuf::UnknownFieldSet const&)'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::NameOfEnum[abi:cxx11](google::protobuf::EnumDescriptor const*, int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::PopLimit(int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::io::CodedInputStream::ReadTagFallback(unsigned int)'
./.libs/libmesos.so: undefined reference to `google::protobuf::FieldDescriptor::type() const'
./.libs/libmesos.so: undefined reference to `google::protobuf::internal::WireFormatLite::WriteString(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, google::protobuf::io::CodedOutputStream*)'
```

**Comment by globin** [comment]
I'll cherry-pick the commits except the last, that is at least the state we had before the build broke

**Comment by globin** [comment]
I've fixed the rest by:
dabb296c76dea4446665e47990653de909ac054f
49e9a26e77b05f26c8d273a4c9836e919ba5878f
c2ce38473b48e3249112698ee89473a662cd50a9

**Pull request closed without merging**